versioning/lockfiles/introduction.rst

```diff
@@ -0,0 +1,240 @@
+
+.. code-block:: json
+
+    {
+        "graph_lock": {
+            "nodes": {
+                "0": {
+                    "ref": "pkgb/0.1@user/testing",
+                    "options": "shared=False",
+                    "requires": ["1"],
+                    "path": "..\\conanfile.py",
+                    "context": "host"
+                },
+                "1": {
+                    "ref": "pkga/0.1@user/testing",
+                    "options": "",
+                    "package_id": "4024617540c4f240a6a5e8911b0de9ef38a11a72",
+                    "prev": "0",
+                    "context": "host"
+                }
+            },
+            "revisions_enabled": false
+        },
+        "version": "0.4",
+        "profile_host": "[settings]\narch=x86_64\narch_build=x86_64\nbuild_type=Release\ncompiler=Visual Studio\ncompiler.runtime=MD\ncompiler.version=15\nos=Windows\nos_build=Windows\n[options]\n[build_requires]\n[env]\n"
+    }
+
+
+We can see the ``pkga/0.1@user/testing`` dependency in the lockfile, together with its ``package_id``. This
+dependency is fully locked. The ``pkgb/0.1@user/testing`` doesn't have a ``package_id`` yet, because so far it
+is just a local *conanfile.py* as a consumer, not a package. But the ``user/testing`` user and channel are already defined.
+
+It is important to note that the *pkgb_deps.lock* lockfile contains the current ``profile`` for the current configuration.
+ 
+At this moment we have captured the dependency graph for ``pkgb``. Now, it would be possible that a new version 
+of ``pkga`` is created:
+
+
+.. code-block:: bash
+
+    # The recipe generates different package code depending on the version, automatically
+    $ conan create pkga pkga/0.2@user/testing
+
+If now we install and build our code in ``pkgb`` we would get:
+
+.. code-block:: bash
+
+    $ mkdir pkgb/build
+    $ cd pkgb/build
+    $ conan install ..
+    > ... pkga/0.2@user/testing from local cache - Cache
+    # Example for VS, use your compiler here
+    $ cmake ../src -G "Visual Studio 15 Win64"
+    $ cmake --build . --config Release
+    $ ./bin/greet
+    HelloA 0.2 Release
+    HelloB Release!
+    Greetings Release!
+
+But as explained above, the purpose of the lockfile is to capture the dependencies, and used them later.
+Lets pass the lockfile as an argument to guarantee the usage of the locked ``pkga/0.1@user/testing`` dependency:
+
+.. code-block:: bash
+
+    $ conan install .. --lockfile=../locks/pkgb_deps.lock
+    > ... pkga/0.1@user/testing from local cache - Cache
+    $ cmake ../src -G "Visual Studio 15 Win64"
+    $ cmake --build . --config Release
+    $ ./bin/greet
+    HelloA 0.1 Release
+    HelloB Release!
+    Greetings Release!
+
+That's it. We managed to depend on ``pkga/0.1@user/testing`` instead of the ``pkga/0.2@user/testing`` even if the later
+satisfy the version range and is available in the cache, because we reproduced the same dependency graph at the time
+the lockfile was captured.
```

Comment:
```suggestion
satisfies the version range and is available in the cache. Using the same dependency was possible because we used the information stored in the lockfile.
```